File: Online_Shop/Online_Shop/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Online_Shop.Models
{
    public class Product
    {
        public int Id { get; set; }

        [Required, MaxLength(30)]
        public string? Name { get; set; }

        [Required]
        public int Price { get; set; }

        [Required]
        public int Amount { get; set; }

        public string? Description { get; set; }

        public byte[]? Image { get; set; }

        public List<OrderProduct>? OrderProducts { get; set; }

        public bool Deleted { get; set; }

        public User? User { get; set; }

        public int UserId { get; set; }

    }
}

File: onlineshop-ui/react-app/src/models/Product.js
export class RegisterModel {
  constructor(
    Id,
    Email,
    Name,
    Price,
    Amount,
    Description,
    Image,
    OrderProducts,
    Deleted,
    User,
    UserId
  ) {
    this.Id = Id;
    this.Email = Email;
    this.Name = Name;
    this.Price = Price;
    this.Amount = Amount;
    this.Description = Description;
    this.Image = Image;
    this.OrderProducts = OrderProducts;
    this.Deleted = Deleted;
    this.Image = Image;
    this.User = User;
    this.UserId = UserId;
  }
}
